module/localisation/BallLocalisation/src/BallLocalisation.cpp

```diff
@@ -34,24 +36,36 @@ namespace module::localisation {
             last_time_update_time        = NUClear::clock::now();
         });
 
-        on<Configuration, Sync<BallLocalisation>>("BallLocalisation.yaml").then([this](const Configuration& config) {
+        on<Configuration, Sync<BallLocalisation>>("BallLocalisation.yaml").then([this](const Configuration& cfg) {
             auto message = std::make_unique<std::vector<Ball>>();
             emit(message);
             emit(std::make_unique<Ball>());
 
-            ball_pos_log = config["ball_pos_log"].as<bool>();
+            log_level = cfg["log_level"].as<NUClear::LogLevel>();
+
+            ball_pos_log = cfg["ball_pos_log"].as<bool>();
             // Use configuration here from file RobotParticleLocalisation.yaml
-            filter.model.n_rogues    = config["n_rogues"].as<int>();
-            filter.model.resetRange  = config["reset_range"].as<Expression>();
-            filter.model.n_particles = config["n_particles"].as<int>();
+            filter.model.n_rogues    = cfg["n_rogues"].as<int>();
+            filter.model.resetRange  = cfg["reset_range"].as<Expression>();
+            filter.model.n_particles = cfg["n_particles"].as<int>();
+
+            config.start_variance = cfg["start_variance"].as<Expression>();
+        });
 
-            const Eigen::Vector2d start_state    = config["start_state"].as<Expression>();
-            const Eigen::Vector2d start_variance = config["start_variance"].as<Expression>();
+        on<Startup, With<FieldDescription>>().then([this](const FieldDescription& fd) {
+            // Left side penalty mark
+            config.start_state.emplace_back((fd.dimensions.field_length / 2.0) + fd.dimensions.penalty_mark_distance,
+                                            (-fd.dimensions.field_width / 2.0));
 
-            // start_variance.asDiagonal() == the starting covariance matrix
-            filter.set_state(start_state, start_variance.asDiagonal());
+            // Right side penalty mark
+            config.start_state.emplace_back((fd.dimensions.field_length / 2.0) + fd.dimensions.penalty_mark_distance,
```

Comment:
Swap x and y vectors and fix y distance ( x = -??width/2, y = length/2 - penaltyDistance)